iotedgedev/utility.py

```diff
@@ -41,6 +41,17 @@ def call_proc(self, params, shell=False):
         except Exception as e:
             self.output.error("Error while executing command: {0}. {1}".format(' '.join(params), str(e)))
 
+    def is_dir_empty(self, name):
+        if name == ".":
```

Comment:
I feel like lines 45-48 where you resolve name to full name shouldn't be in this method called is_dir_empty.  I think that logic can be in the create method for now.

Reply:
Done. I'd actually keep that logic in is_empty_dir because although its a utility method I think it can well encapsulate the actual name-to-path manipulation, but that's just an implementation detail really.